pages/api/banks/v1/index.js

```diff
@@ -4,8 +4,21 @@ import { getBanksData } from '@/services/banco-central';
 const action = async (request, response) => {
   const allBanksData = await getBanksData();
 
-  response.status(200);
-  response.json(allBanksData);
+  if (request.query.search) {
+    const { search } = request.query;
+
+    const filteredBanks = allBanksData.filter(
+      ({ code, name }) =>
+        String(code).toLowerCase().includes(search.toLowerCase()) ||
+        String(name.toLowerCase()).includes(search.toLowerCase())
```

Comment:
Tenho duas sugestões:
-  tirar a chamada `search.toLowerCase()` de dentro do _callback_ e usar uma variável `const lowerCaseSearch` ou algum nome que achar mais adequado;
- a primeira vista a ordem de chamada diferente de `String()` e `.toLowerCase()` parece erro de digitação, mas lendo esse código é possível entender que `code` seja numérico, se for isso mesmo seria interessante adicionar comentários explicando a necessidade da diferença

Reply:
Sim, o code retorna any (mas vi que tipicamente é um number). Vou inserir o comentário